NDTensors/src/blocksparse/contract.jl

```diff
@@ -1,12 +1,14 @@
 using .AlgorithmSelection: Algorithm, @Algorithm_str
 
 function contract(
-  tensor1::BlockSparseTensor,
+  tensor1::Exposed{<:AbstractArray,<:BlockSparseTensor},
```

Comment:
Modifying this code to look like this enables one to utilize cuTENSOR for `BlockSparseTensors`. However currently I don't believe there is an easy way to convert the blocksparsetensors into cutensors. So in the cuTENSOR code I have a dispatched function which just uses CUDA's `mul!` function. But I am working on a solution

Reply:
I am removing this and moving the CuTensor portion to the inplace `contract!` function